File: training-data/rust/quicksort.rs
fn quicksort(items: &[i32]) -> Vec<i32> {
    if items.len() <= 1 {
        return items.to_vec();
    }
    let pivot = items[items.len() / 2];
    let left: Vec<i32> = items.iter().filter(|&&x| x < pivot).cloned().collect();
    let mid: Vec<i32> = items.iter().filter(|&&x| x == pivot).cloned().collect();
    let right: Vec<i32> = items.iter().filter(|&&x| x > pivot).cloned().collect();

    let mut result = quicksort(&left);
    result.extend(mid);
    result.extend(quicksort(&right));
    result
}

fn main() {
    let data = vec![5, 3, 8, 1, 9, 2];
    println!("{:?}", quicksort(&data));
}

File: training-data/dart/quicksort.dart
List<int> quicksort(List<int> items) {
  if (items.length <= 1) return items;
  final pivot = items[items.length ~/ 2];
  final left = items.where((x) => x < pivot).toList();
  final mid = items.where((x) => x == pivot).toList();
  final right = items.where((x) => x > pivot).toList();
  return [...quicksort(left), ...mid, ...quicksort(right)];
}

void main() {
  print(quicksort([5, 3, 8, 1, 9, 2]));
}
